### PR TITLE
[Resources] Add Deployment scripts as a new resource type - preview -

### PR DESCRIPTION
**Pull request by filizt**
Here's the link to the approved swagger: Azure/azure-rest-api-specs-pr#924

Closing the PR which was created against DeploymentScripts branch.
https://github.com/Azure/azure-rest-api-specs/pull/7485

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/resources/resource-manager/Microsoft.Resources/preview/2019-10-01-preview/deploymentScripts.json [hunk 1]
```diff
@@ -0,0 +1,892 @@
...
+          }
+        },
+        "scriptContent": {
+          "type": "string",
+          "maxLength": 32000,
+          "description": "Script body."
+        },
+        "arguments": {
+          "type": "string",
+          "description": "Command line arguments to pass to the script. Arguments are seperated by spaces. ex: -Name blue* -Location 'West US 2' "
```
Comment (nschonni) [inline specification/resources/resource-manager/Microsoft.Resources/preview/2019-10-01-preview/deploymentScripts.json]:
```suggestion
          "description": "Command line arguments to pass to the script. Arguments are separated by spaces. ex: -Name blue* -Location 'West US 2' "
```

**Comment by Juliehzl** [comment]
It is already approved in private repo. LGTM. @filizt can you try to make CI pass?

**Review by Juliehzl — APPROVED** [review]

**Comment by filizt** [comment]
@Juliehzl the swagger is already approved by ARM. See the private repo RP here: https://github.com/Azure/azure-rest-api-specs-pr/pull/924. Could you merge please?

File: specification/resources/resource-manager/readme.csharp.md [hunk 2]
```diff
@@ -23,6 +23,7 @@ batch:
   - package-resources: true
   - package-subscriptions: true
   - package-links: true
+  - package-package-deploymentscripts: true
```
Comment (isra-fel) [inline specification/resources/resource-manager/readme.csharp.md]:
```suggestion
  - package-deploymentscripts: true
```

File: specification/resources/resource-manager/readme.md [hunk 3]
```diff
@@ -452,6 +464,26 @@ directive:
     from: managedapplications.json
     where: $.definitions.GenericResource.properties
     reason: managedBy is a top level property
+  - from: deploymentScripts.json
+    suppress: TrackedResourceGetOperation 
+    where: $.definitions.AzureCliScript
+    reason: Tooling issue.
```
Comment (nschonni) [inline specification/resources/resource-manager/readme.md]: Curious what the actual tool issue is

Comment (filizt) [reply specification/resources/resource-manager/readme.md]: @nschonni we asked about the linter errors to the swagger team and they directed us to suppress the errors. Below is the copy of the email thread.

________________________________________
From: Filiz Topatan <Filiz.Topatan@microsoft.com>
Sent: Thursday, September 19, 2019 6:47 AM
To: Ruoxuan Wang; Zhiqing Qiu; Marcin Jastrzebski; Lei Wang (AZURE)
Cc: Catherine Zhu; Marc Reyhner
Subject: RE: AutoRest Linter Errors on Deployment Scripts Swagger
 
+ Marc, Marcin, Lei
 
Thanks for looking into this Ruoxuan. I’ll add the suppression accordingly.
 
From: Ruoxuan Wang <Ruoxuan.Wang@microsoft.com> 
Sent: Thursday, September 19, 2019 1:14 AM
To: Filiz Topatan <Filiz.Topatan@microsoft.com>; Zhiqing Qiu <Zhiqing.Qiu@microsoft.com>
Cc: Catherine Zhu <yungez@microsoft.com>
Subject: RE: AutoRest Linter Errors on Deployment Scripts Swagger
 
Hi, @Filiz Topatan
 
After syncing up with Zhiqing offline. The errors you mentioned are actually false positive. You can add suppression section in `readme.md`.  Such as
 
``` yaml
directive:
  - suppress: TrackedResourceGetOperation
    reason: <reason here>
    where:
                - $.definitions.CliScript
  - suppress: TrackedResourcePatchOperation
reason: <reason here>
where:
                - $.definitions.PowerShellScript
```
 
After debug it. The reason is complex. The rule “TrackedResourceGetOperation” and “TrackedResourcePatchOperation” not only check whether existing “get” or “patch” operation but also check “response.schema.reference” whether equal with name of definition. In your case, The name of CLIScript and PowerShellScript is different from the “response.schema.reference”. So the lint tool report the errors.
 
Thanks,
Ruoxuan
From: Filiz Topatan <Filiz.Topatan@microsoft.com>
Sent: Thursday, September 19, 2019 10:46 AM
To: Ruoxuan Wang <Ruoxuan.Wang@microsoft.com>; Zhiqing Qiu <Zhiqing.Qiu@microsoft.com>
Cc: Catherine Zhu <yungez@microsoft.com>
Subject: RE: AutoRest Linter Errors on Deployment Scripts Swagger
 
I already talked to Ruoxuan about the below issue and he mentioned he’s new in the area and so wanted me to add Zhiqing Qui to the email to get some help on the issue.
 
From: Filiz Topatan 
Sent: Wednesday, September 18, 2019 7:29 PM
To: Ruoxuan Wang <Ruoxuan.Wang@microsoft.com>; Zhiqing Qiu <Zhiqing.Qiu@microsoft.com>
Cc: Catherine Zhu <yungez@microsoft.com>
Subject: AutoRest Linter Errors on Deployment Scripts Swagger
 
Hi,
 
We have a swagger for a new resource type called deploymentScript in Microsoft.Resources namespace (we haven't yet created a PR for this new swagger). We are getting AutoRest linter warnings and errors during the validation however we couldn't figure out the reason for these errors. 
 
We suspect that these warnings and errors are somehow related to us using the discriminator pattern where we derive CliScript and PowerShellScript objects from the base class DeploymentScripts. The errors are due to not having Patch or Get operations for these derived types. It's as if the tool is not seeing the polymorphic relationship between the base class and it's derived classes where both Path and Get (and other) operations are defined. It'd be great if you can provide some input as to why these errors are generated and if there is an error on our side or a tooling issue causing them.
 
There are a few known/expected errors, I'm disregarding those and only include related errors in the attached LinterErrors file. Please see other attached files for the swagger and request example
 
We also have an ARM review of the swagger early next week, if it’s possible to shed a light on this issue, it is appreciated as we might get questions on it during the review.
 
Let me know if you need more information on the errors or our design.
 
Thanks
Filiz

Comment (nschonni) [reply specification/resources/resource-manager/readme.md]: Thanks! I just like kicking the tires on the tools, and like seeing where the limitations pop up

**Review by KrisBash — APPROVED** [review]
PR was previously reviewed in private repo

**Comment by yungezz** [comment]
@Juliehzl  could you  pls merge the PR? thanks.

File: specification/resources/resource-manager/readme.md [hunk 4]
```diff
@@ -56,6 +56,18 @@ tag: package-links-2016-09
 tag: package-managedapplications-2018-06
 ```
 
+``` yaml $(package-deploymentscripts)
+tag: package-deploymentscripts-2019-10-preview
+```
+
+### Tag: package-resources-2019-10-preview
```
Comment (jiasli) [inline specification/resources/resource-manager/readme.md]: Shouldn't this be `package-deploymentscripts-2019-10-preview`?

Comment (filizt) [reply specification/resources/resource-manager/readme.md]: Tag name change is fine with us. It sounds like it'll be more clear this way.

**Pull request merged**